src/libraries/Microsoft.Extensions.Logging.EventSource/ref/Microsoft.Extensions.Logging.EventSource.cs

```diff
@@ -17,7 +17,7 @@ namespace Microsoft.Extensions.Logging.EventSource
     public partial class EventSourceLoggerProvider : Microsoft.Extensions.Logging.ILoggerProvider, System.IDisposable
     {
         public EventSourceLoggerProvider(Microsoft.Extensions.Logging.EventSource.LoggingEventSource eventSource) { }
-        public Microsoft.Extensions.Logging.ILogger CreateLogger(string categoryName) { throw null; }
+        public Microsoft.Extensions.Logging.ILogger CreateLogger(string? categoryName) { throw null; }
```

Comment:
`ILoggerProvider` has categoryName as non-nullable. So this should be `string categoryName`.

We don't want to allow this to be null.

Reply:
Should `EventSourceLogger.CategoryName` be nullable then? Maybe it doesn't really matter sines it's an internal API, and nothing breaks when it's nullable, but still.

Reply:
> Should EventSourceLogger.CategoryName be nullable then? 

I would say `no`. If nothing ever creates it with a `null` name, then I wouldn't make it nullable.